prime-angular-frontend/src/app/modules/enrolment/pages/self-declaration/self-declaration.component.ts

```diff
@@ -100,9 +102,15 @@ export class SelfDeclarationComponent extends BaseEnrolmentProfilePage implement
   }
 
   public onSubmit() {
-    const hasBeenThroughTheWizard = true;
-    this.hasAttemptedFormSubmission = true;
-    super.onSubmit(hasBeenThroughTheWizard);
+    if (!this.clickedDownloadDoc) {
+      const hasBeenThroughTheWizard = true;
+      this.hasAttemptedFormSubmission = true;
+      super.onSubmit(hasBeenThroughTheWizard);
+    } else {
+      // If downloading previously uploaded doc, remain on page.
+      // Reset flag.
+      this.clickedDownloadDoc = false;
+    }
```

Comment:
when the enrolment is editable and enrollee tries to edit the self-declaration on overview page by the pencil icon, uploading a different file would not change the file button name.
e.g. during initial enrolment, upload a `file1.png` for one of the self-declaration.
after making the enrolment editable, upload a `file2.png' to replace the existing `file1.png'. The download button remains `file1.png' in overview page.

Reply:
@johnnyl6  Not sure I understand what you expect:  In `dev` and this PR, uploading another self-declaration document **adds** the document to the ones already associated with that self-declaration question.

Reply:
Actually might be 2 issues:
1. Changed attachment to `File2.png' but the download button still says the old one `Capture.PNG`
![image](https://user-images.githubusercontent.com/58282608/146588651-caf41c23-9737-48de-a64f-35c1e63e7d1a.png)

2. Changed selection to `No` but download button is still visible.
![image](https://user-images.githubusercontent.com/58282608/146588700-26c2b878-4155-42d1-b752-2b4126d48c7b.png)